Pspcl/Pspcl.Web/Controllers/AddStockController.cs

```diff
@@ -1,12 +1,25 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using Pspcl.DBConnect;
+using Pspcl.Services;
+using Pspcl.Web.Models;
 
 namespace Pspcl.Web.Controllers
 {
-    public class AddStockController : Controller
+    
+    public class AddStockController: Controller
     {
-        public IActionResult AddStock()
+        private readonly StockMappingService _stockMappingService;
+        public AddStockController(StockMappingService stockMappingService)
         {
-            return View();
+            _stockMappingService = stockMappingService;
+        }
+        public IActionResult AddStock(Model model)
```

Comment:
If this is a get action, why we are passing model as parameter, please clear your concepts about action and their types